### PR TITLE
enable kubelet server cert bootstrapping

### PR DESCRIPTION
**Pull request by sjenning**
@smarterclayton @ericavonb @mrogers950 @rphillips 

wait for https://github.com/openshift/cluster-machine-approver/pull/3 to merge and ensure that machine-approver is running

### CODE REVIEW

**Comment by sjenning** [comment]
Test cluster is working with the approver starting and approving new machines as expected, but I think I need to move the approver to runlevel 0.  At runlevel 1 is starts late and kubelets will not accept incoming connections until the server cert is installed. https://github.com/openshift/cluster-machine-approver/pull/4

File: templates/_base/master/files/kubelet.yaml [hunk 1]
```diff
@@ -13,3 +13,6 @@ contents:
     runtimeRequestTimeout: 10m
     serializeImagePulls: false
     staticPodPath: /etc/kubernetes/manifests
+    featureGates:
+      RotateKubeletServerCertificate: true
```
[the remark this answers is not in the document]
Comment (sjenning) [reply templates/_base/master/files/kubelet.yaml]: Yes, this is set for workers, but not masters.  I didn't want to make that change as part of this until I understood why this was done.

**Comment by smarterclayton** [comment]
I think you'll need to ensure the signer is running early in bootstrap.

**Comment by sjenning** [comment]
i think openshift/cluster-machine-approver#4 may have not gone in before the previous retest
/retest

File: templates/_base/worker/files/kubelet.yaml [hunk 2]
```diff
@@ -14,3 +14,6 @@ contents:
     runtimeRequestTimeout: 10m
     serializeImagePulls: false
     staticPodPath: /etc/kubernetes/manifests
+    featureGates:
+      RotateKubeletServerCertificate: true
+    serverTLSBootstrap: true
```
Comment (rphillips) [inline templates/_base/worker/files/kubelet.yaml]: EOL got stripped on this file

**Comment by sjenning** [comment]
needs https://github.com/openshift/origin/pull/21551 plus a PR to the cluster-network-operator to not set the kubelet serving info in the generated NodeConfig.

**Comment by smarterclayton** [comment]
Please rebase - this is critical for getting metrics from nodes.

**Pull request merged**